### DISCUSSION TITLE
Role Based Column Security

### DISCUSSION THREAD

**Discussion by onurc4kir** [supabase/supabase#10827] [Questions]
Hi,
I have profiles table like: 
id , role, is_verified, comment
1 , admin, true,comment1
2, customer, false,comment 2

I want admin role to have access to change everything on profiles table but I also want customer role to view all columns and edit only comment column. 

How can I make this by using RLS or triggers ?

**Comment by GaryAustin1** [comment]
You cannot with RLS.

You can do that with a before update trigger function.  You just set new.col = old.col  and use an if statement (case if SQL) to determine your type of use.

You can also use Postgres column grants, but only if you have Postgres roles established and use able.
https://www.postgresql.org/docs/14/sql-grant.html
